docs/specs/phasing-vector.md

```diff
@@ -0,0 +1,254 @@
+    --output phased.vcf.gz
+```
+
+Notes:
+
+* SHAPEIT4 can be run on a specific genome region (``--region``),
+  allowing some parallelisation by genome region. @@TODO discuss
+  appropriate region size.
+
+* A genetic map file is needed (``--map``). We have one for
+  *An. gambiae*. For other species where no map is available a map can
+  *be created assuming constant recombination rate.
+
+* SHAPEIT4 can use multi-threading (``--threads``) which may be
+  important especially when phasing larger cohorts.
+
+* SHAPEIT4 allows tuning of the MCMC iteration scheme
+  (``--mcmc-iterations``), which allows the tradeoff between accuracy
+  and speed to be explored, see
+  [documentation](https://odelaneau.github.io/shapeit4/#documentation)
+  point 5. The default is ``5b,1p,1b,1p,1b,1p,5m`` but an alternative
+  offering greater accuracy is suggested
+  ``10b,1p,1b,1p,1b,1p,1b,1p,10m``. @@TODO spike to get a sense of
+  performance cost of increasing number of burn-in and/or main
+  iterations.
+
+* SHAPEIT4 allows tuning of the number of conditioning haplotypes via
+  the ``--pbwt-depth`` parameter, with default 4 and higher values
+  providing more accuracy at slower performance. @@TODO spike
+  performance cost of increasing.
+
+* SHAPEIT4 allows tuning of the size of the window used for phasing
+  (``--window``) which is given in cM, default 2.5. @@TODO check this
+  is sensible for *An. gambiae*.
+
+* Parameters may need to be tuned to adjust for high variant
+  density. The option ``--sequencing`` alters default parameters for
+  PBWT and IBD2 mapping to be appropriate for sequencing data, and so
+  should be a good baseline to start from, but some tuning of
+  individual parameters may still be needed. @@TODO figure out how
+  --sequencing changes defaults for other params.
+
+* The ``--reference`` parameter is optional and given only if phasing
+  against a reference panel.
+
+* The parameter ``--use-PS`` tells SHAPEIT4 to use the phasing
+  information provided by WhatsHap.
+
+
+#### Step: Ligate regions?
+
+@@TODO If SHAPEIT4 is run in separate genome regions, do we need to
+ligate the regions back together? This was previously needed for
+SHAPEIT2 but I can't see anything about this on the SHAPEIT4 docs. Or
+should we instead phase entire chromosomes at a time?
+
+Output should be a whole-genome VCF, or one VCF per chromosome also
+acceptable.
+
+
+#### Step: VCF to Zarr
+
+For ease of downstream analysis, convert the final VCF to Zarr via the
+scikit-allel vcf_to_zarr function.
+
+```
+@@TODO function params
+```
+
+N.B., there is no need to zip the outputs of this step, we will use
+them unzipped.
+
+
+## Implementation notes
+
+@@TODO something about chunking and parallelisation
+
+* SHAPEIT4 has optimisations to use AVX2 instruction set, and so
+  compiling and running the program on CPUs supporting AVX2 is
+  preferable. See also [SHAPEIT4 installation
+  notes](https://odelaneau.github.io/shapeit4/#installation).
+
```

Comment:
Open question: how to handle missing genotype calls? Reasonable to impute, or should be left as missing? If want to leave as missing, does this mean we need to post-process data, if SHAPEIT4 always performs imputation?